into/backends/h5py.py

```diff
@@ -31,16 +33,8 @@ def discover_h5py_dataset(d):
     return dshape(s.replace('object', 'string'))
 
 
-def varlen_dtype(dt):
-    """ Inject variable length string element for 'O' """
-    if "'O'" not in str(dt):
-        return dt
-    varlen = h5py.special_dtype(vlen=unicode)
-    return np.dtype(eval(str(dt).replace("'O'", 'varlen')))
-
-
```

Comment:
`dtype("O")` can have different metadata.  In particular, h5py accepts a dtype("O") that it specifically interprets as variable length string.

http://docs.h5py.org/en/latest/special.html
http://docs.h5py.org/en/latest/strings.html

```Python
In [1]: import h5py

In [2]: h5py.special_dtype(vlen=str)
Out[2]: dtype('O')
```

It's been a while since I dealt with this though, so my recollection is fuzzy.